### ISSUE TITLE
Strange SC1106 error

### ISSUE THREAD

**Issue by fdellwing** [koalaman/shellcheck#1836]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1106
- My shellcheck version (`shellcheck --version` or "online"): 0.7
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
_delete=$((echo -ne $_command) | /usr/bin/sshpass -p $_password sftp $_username@$_server)

```

#### Here's what shellcheck currently says:
```
 In etc/scripts/backup.sh line 111:
            _delete=$((echo -ne $_command) | /usr/bin/sshpass -p $_password sftp $_username@$_server)
                            ^-- SC1106: In arithmetic contexts, use != instead of -ne
```

#### Here's what I wanted or expected to see:
Nothing (Running shellcheck with -S error)

**Comment by Nightfirecat** [comment]
Looks like shellcheck is incorrectly assuming that `$((` here denotes the start of an arithmetic operation.

As an aside, is your snippet not the same as the following?

```sh
#!/bin/bash
_delete=$(/usr/bin/sshpass -p $_password sftp $_username@$_server <<< "$_command")
```

If so, this would help you avoid the false positive SC1106 in your script for now.

**Comment by koalaman** [comment]
There's a more informative message being suppressed:

        ^-- SC1102: Shells disambiguate $(( differently or not at all. For $(command substition), add space after $( . For $((arithmetics)), fix parsing errors.

I just went with the pragmatic workaround of bumping that from a warning to an error so it's easier to see why ShellCheck is interpreting this incorrectly and what to do about it.

Obviously correct code shouldn't trigger and it would be simple enough to throw away any errors relating to arithmetic expressions in this kind of construct, but this would again complicate things for people with the opposite problem of trying to figure out why it's *not* being treated as a `$((..))`

**Comment by fdellwing** [comment]
It's a legacy script, so it will be rewritten in the future. Thanks anyway for solving the obvious problem.
